src/axios-adapter.ts

```diff
@@ -0,0 +1,76 @@
+import * as runtime from './runtime';
+import * as FormData from 'form-data';
+import * as assert from 'assert';
+import axios, { AxiosResponse } from 'axios';
+
+function toRequestData(data: runtime.server.RequestBody<any> | undefined) {
+  if (data == null) {
+    return data;
+  }
+  if (data.contentType === 'application/json') {
+    return axiosToJson(data.value);
+  }
+  if (['application/x-www-form-urlencoded', 'multipart/form-data'].indexOf(data.contentType) >= 0) {
+    const form = new FormData();
+    Object.keys(data.value).forEach(key => {
+      const element = data.value[key];
+      if (element instanceof runtime.make.FormBinary) {
+        form.append(key, element.binary, element.options);
+      } else {
+        form.append(key, element);
+      }
+    });
+    return form;
+  }
+  assert.fail('unknown content type for axios client ' + data.contentType);
+}
+
+function axiosToJson(data: any) {
+  if (data instanceof runtime.valueClass.ValueClass) {
+    return runtime.valueClass.toJSON(data);
+  }
+  return data;
+}
+
+export const bind: runtime.client.ClientAdapter = async (
```

Comment:
Maybe unnecessary, but having a base client (and server) adapter from which axios/koa can "inherit" would be nice to help extend oats to use different client/servers in the future (native node http? express?) 🤔

Reply:
what would the base adapter do?

Reply:
Nothing. It'd be more of an interface sort of thing...
Seen as this only exports a single function, it may very well be unnecessary 😅